xds/src/main/java/io/grpc/xds/EnvoyProtoData.java

```diff
@@ -135,17 +141,370 @@ public String toString() {
   }
 
   /**
-   * See corresponding Envoy proto message {@link io.envoyproxy.envoy.api.v2.core.Locality}.
+   * See corresponding Envoy proto message {@link io.envoyproxy.envoy.config.core.v3.Node}.
+   */
+  public static final class Node {
+
+    private final String id;
+    private final String cluster;
+    @Nullable
+    private final Map<String, ?> metadata;
+    @Nullable
+    private final Locality locality;
+    private final List<Address> listeningAddresses;
+    private final String buildVersion;
+    private final String userAgentName;
+    @Nullable
+    private final String userAgentVersion;
+    private final List<String> clientFeatures;
+
+    private Node(
+        String id, String cluster, @Nullable Map<String, ?> metadata, @Nullable Locality locality,
+        List<Address> listeningAddresses, String buildVersion, String userAgentName,
+        @Nullable String userAgentVersion, List<String> clientFeatures) {
+      this.id = id;
```

Comment:
How the nullity for these fields are determined? Some of them have `checkNotNull` while some others don't.

I don't see the different between `id` and `cluster` while one of them has `checkNotNull` and the other doesn't. `Node`'s builder has a default value of `""` for `id` and its setter can never be `null`.